vector/src/main/java/im/vector/app/core/di/ActiveSessionHolder.kt

```diff
@@ -80,18 +87,27 @@ class ActiveSessionHolder @Inject constructor(
     }
 
     fun hasActiveSession(): Boolean {
-        return activeSessionReference.get() != null
+        return activeSessionReference.get() != null || authenticationService.hasAuthenticatedSessions()
     }
 
     fun getSafeActiveSession(): Session? {
-        return activeSessionReference.get()
+        return runBlocking { getOrInitializeSession(startSync = true) }
```

Comment:
this is the crash fix, we'll always attempt to initialise the session (including perform a migration if needed)

the hope is that most users will only perform a migration via the `MainActivity` flow (tapping the launcher icon/notification) which will give them a loading bar, however it's possible the ROMs/OEMs can restore the activity and application directly